In [1]:
from transformers import BertForMaskedLM,pipeline

In [2]:
bert_model = BertForMaskedLM.from_pretrained('bert-base-cased')

config.json:   0%|          | 0.00/570 [00:00<?, ?B/s]

model.safetensors:   0%|          | 0.00/436M [00:00<?, ?B/s]

Some weights of the model checkpoint at bert-base-cased were not used when initializing BertForMaskedLM: ['cls.seq_relationship.bias', 'bert.pooler.dense.weight', 'bert.pooler.dense.bias', 'cls.seq_relationship.weight']
- This IS expected if you are initializing BertForMaskedLM from the checkpoint of a model trained on another task or with another architecture (e.g. initializing a BertForSequenceClassification model from a BertForPreTraining model).
- This IS NOT expected if you are initializing BertForMaskedLM from the checkpoint of a model that you expect to be exactly identical (initializing a BertForSequenceClassification model from a BertForSequenceClassification model).


In [3]:
print(bert_model)

BertForMaskedLM(
  (bert): BertModel(
    (embeddings): BertEmbeddings(
      (word_embeddings): Embedding(28996, 768, padding_idx=0)
      (position_embeddings): Embedding(512, 768)
      (token_type_embeddings): Embedding(2, 768)
      (LayerNorm): LayerNorm((768,), eps=1e-12, elementwise_affine=True)
      (dropout): Dropout(p=0.1, inplace=False)
    )
    (encoder): BertEncoder(
      (layer): ModuleList(
        (0-11): 12 x BertLayer(
          (attention): BertAttention(
            (self): BertSelfAttention(
              (query): Linear(in_features=768, out_features=768, bias=True)
              (key): Linear(in_features=768, out_features=768, bias=True)
              (value): Linear(in_features=768, out_features=768, bias=True)
              (dropout): Dropout(p=0.1, inplace=False)
            )
            (output): BertSelfOutput(
              (dense): Linear(in_features=768, out_features=768, bias=True)
              (LayerNorm): LayerNorm((768,), eps=1e-12, elementwise_a

Postinal embedding has 512 as max sequence length of bert base is 512 
Token_type has 2 as there can upto 2 sentences

**While bert uncased has 30522 tokens , Bert Cased has 28996 tokens. Here in MLM we mask about 15% of corpus for bert to predict. 
In this after feedforward net and Softmax we get tokens with highest probabiltiy to fit in the blank******

# Example

In [21]:
ex = pipeline("fill-mask",model='bert-base-cased')

Exception: Impossible to guess which tokenizer to use. Please provide a PreTrainedTokenizer class or a path/identifier to a pretrained tokenizer.

In [18]:
pred = ex(f'Mumbai is a great {ex.tokenizer.mask_token} to live  in.')

In [19]:
for p in pred:
    print(f"Token:{p['token_str']}. Score:{100*p['score']}")

Token:city. Score:57.43005871772766
Token:place. Score:33.80904495716095
Token:town. Score:0.9498416446149349
Token:environment. Score:0.7329258602112532
Token:country. Score:0.7319500669836998
